ui/src/pages/AllStories/index.tsx

```diff
@@ -306,25 +324,40 @@ export const AllStories: React.FC = () => {
   if (!allStories) return <div>Loading all stories table..</div>;
   return (
     <>
+      <Prompt
+        message={(location) =>
+          `Are you sure you want to go to ${location.pathname}? Your changes on this page will not be saved.`
```

Comment:
Wait shouldn't we only conditionally render this prompt? Right now we render it everytime, regardless of the page condition.

Reply:
Set a boolean variable that becomes true whenever a switch or checkbox gets toggled?

Reply:
Yeah that makes sense to me

Reply:
Actually we can just do pending changes length > 0